src/index.ts

```diff
@@ -283,12 +283,24 @@ const getData = async (
 };
 
 const calcRankingOfDay = (data: Array<AggregatedData>): Array<Ranking> => {
+  let prevMessageCount: number;
+  let prevRank: number;
```

Comment:
I want to merge these defined variables into a map function for better readability.
For example:

```ts
  const calcRank = (() => {
    let prevMessageCount: number;
    let prevRank: number;
    return (channel: AggregatedData, i: number) => {
      const rank = channel.messages.length === prevMessageCount
        ? prevRank
        : i + 1;

      prevMessageCount = channel.messages.length;
      prevRank = rank;

      return {
        channel,
        rank,
      };
    };
  })();

  return data
    .sort((a, b) => b.messages.length - a.messages.length)
    .map(calcRank);
```

Reply:
Thanks, I'll fix it that way.

Reply:
Fixed: 7cf9c1af2a357b5458d59490c1239908748a0fd2

Reply:
Thank you!